src/common/kvcache_mgr.h

```diff
@@ -77,7 +82,12 @@ class KVCacheMgrImpl : public KVCacheMgrImplBase {
             cache = freeCaches.back();
             freeCaches.pop_back();
         } else {
-            cache = new KVCacheTensor<T>[2 * layers];
+            cache = new KVCacheTensor<T>[2 * layers_];
+        }
+
+        auto maxLen = maxSeqLen > 0 ? maxSeqLen : this->maxSeqLen_;
```

Comment:
Check whether maxSeqLen is larger than this->maxSeqlen

Reply:
If maxSeqLen is set, will always use such length (whether if it is bigger or not). Otherwise have to use the default value.